test/module/irohad/ametsuchi/block_query_test.cpp

```diff
@@ -209,8 +212,148 @@ TEST_F(BlockQueryTest, GetTransactionsWithInvalidTxAndValidTx) {
+  auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getBlocks(1000, 1), 0);
+  wrapper.subscribe();
+  ASSERT_TRUE(wrapper.validate());
+}
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test AND 1 tx created by user2@test
+ * @when height=1, count=1
+ * @then returned exactly 1 block
+ */
+TEST_F(BlockQueryTest, GetExactlyOneBlock) {
+  auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getBlocks(1, 1), 1);
+  wrapper.subscribe();
+  ASSERT_TRUE(wrapper.validate());
+}
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test AND 1 tx created by user2@test
+ * @when count=0
+ * @then no blocks returned
+ */
+TEST_F(BlockQueryTest, GetBlocks_Count0) {
+  auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getBlocks(1, 0), 0);
+  wrapper.subscribe();
+  ASSERT_TRUE(wrapper.validate());
+}
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test AND 1 tx created by user2@test
+ * @when get zero block
+ * @then no blocks returned
+ */
+TEST_F(BlockQueryTest, GetZeroBlock) {
+  auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getBlocks(0, 1), 0);
+  wrapper.subscribe();
+  ASSERT_TRUE(wrapper.validate());
+}
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test AND 1 tx created by user2@test
+ * @when get all blocks starting from 1
+ * @then returned all blocks (2)
+ */
+TEST_F(BlockQueryTest, GetBlocksFrom1) {
+  auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getBlocksFrom(1), blocks_total);
+  size_t counter = 1;
+  wrapper.subscribe([&counter](Block b) {
+    // wrapper returns blocks 1 and 2
+    ASSERT_EQ(b.height, counter++)
+        << "block height: " << b.height << "counter: " << counter;
+  });
+  ASSERT_TRUE(wrapper.validate());
+}
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test AND 1 tx created by user2@test. Block #1 is filled with trash data (NOT JSON).
+ * @when read block #1
+ * @then get no blocks
+ */
+TEST_F(BlockQueryTest, GetBlockButItIsNotJSON) {
+  namespace fs = boost::filesystem;
+  size_t block_n = 1;
+
+  // write something that is NOT JSON to block #1
+  auto block_path = fs::path{block_store_path} / FlatFile::id_to_name(block_n);
+  fs::ofstream block_file(block_path);
+  std::string content = R"(this is definitely not json)";
+  block_file << content;
+  block_file.close();
+
+  auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getBlocks(block_n, 1), 0);
+  wrapper.subscribe();
+
+  ASSERT_TRUE(wrapper.validate());
+}
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test AND 1 tx created by user2@test. Block #1 is filled with trash data (NOT JSON).
+ * @when read block #1
+ * @then get no blocks
+ */
+TEST_F(BlockQueryTest, GetBlockButItIsInvalidBlock) {
+  namespace fs = boost::filesystem;
```

Comment:
Maybe extract to global space?

Reply:
Nop. Maybe in other tests I don't want to use fs as boost::filesystem?